rust/debian/Dockerfile.base

```diff
@@ -47,7 +47,10 @@ RUN DEPS="ca-certificates curl file m4 make locales pkg-config xz-utils cmake \
   && apt-get install -y --no-install-recommends $DEPS > /dev/null \
   && echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen \
   && locale-gen en_US.UTF-8 \
-  && /usr/sbin/update-locale LANG=en_US.UTF-8
+  && /usr/sbin/update-locale LANG=en_US.UTF-8 \
+  && apt-get autoremove -y > /dev/null \
+  && apt-get clean > /dev/null \
+  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/doc
```

Comment:
rm -fr /var/lib/apt/lists/{apt,dpkg,cache,log}

Reply:
DRY - cleanup is still better be done it once at the end. easier to read and maintain.

Reply:
additional image layer is not that bad.

Reply:
It's not the additional layer, it's the fact that the layers that havn't been cleaned up have an extra couple of hundred Mb of data in them as they weren't cleaned up, which persists into the final build image

Reply:
if it is extra hundred of Mb then it is not good.

Reply:
For some data points:

Without these changes:
```
REPOSITORY                   TAG                                   [...]     SIZE
logdna/build-images          rust-bullseye-1-stable-base-x86_64    [...]    1.91GB
```

With these changes:
```
REPOSITORY                   TAG                                   [...]     SIZE
logdna/build-images          rust-bullseye-1-stable-base-x86_64    [...]     2.03GB
```
So it's over 100 MB smaller to do the cleanup within the command than outside

Reply:
right, 10%.